### PR TITLE
feat(platform): add generic ask_question copilot tool

### PR DESCRIPTION
**Pull request by majdyz**
### Why / What / How

**Why:** The copilot can ask clarifying questions in plain text, but that text gets collapsed into hidden "reasoning" UI when the LLM also calls tools in the same turn. This makes clarification questions invisible to users. The existing `ClarificationNeededResponse` model and `ClarificationQuestionsCard` UI component were built for this purpose but had no tool wiring them up.

**What:** Adds a generic `ask_question` tool that produces a visible, interactive clarification card instead of collapsible plain text. Unlike the agent-generation-specific `clarify_agent_request` proposed in #12601, this tool is workflow-agnostic — usable for agent building, editing, troubleshooting, or any flow needing user input.

**How:** 
- Backend: New `AskQuestionTool` reuses existing `ClarificationNeededResponse` model. Registered in `TOOL_REGISTRY` and `ToolName` permissions.
- Frontend: New `AskQuestion/` renderer reuses `ClarificationQuestionsCard` from CreateAgent. Registered in `CUSTOM_TOOL_TYPES` (prevents collapse into reasoning) and `MessagePartRenderer`.
- Guide: `agent_generation_guide.md` updated to reference `ask_question` for the clarification step.

### Changes 🏗️

- **`copilot/tools/ask_question.py`** — New generic tool: takes `question`, optional `options[]` and `keyword`, returns `ClarificationNeededResponse`
- **`copilot/tools/__init__.py`** — Register `ask_question` in `TOOL_REGISTRY`
- **`copilot/permissions.py`** — Add `ask_question` to `ToolName` literal
- **`copilot/sdk/agent_generation_guide.md`** — Reference `ask_question` tool in clarification step
- **`ChatMessagesContainer/helpers.ts`** — Add `tool-ask_question` to `CUSTOM_TOOL_TYPES`
- **`MessagePartRenderer.tsx`** — Add switch case for `tool-ask_question`
- **`AskQuestion/AskQuestion.tsx`** — Renderer reusing `ClarificationQuestionsCard`
- **`AskQuestion/helpers.ts`** — Output parsing and animation text

#### For code changes:
- [x] I have clearly listed my changes in the PR description
- [x] I have made a test plan
- [x] I have tested my changes according to the test plan:
  - [x] Backend format + pyright pass
  - [x] Frontend lint + types pass
  - [x] Pre-commit hooks pass
  - [ ] Manual test: copilot uses `ask_question` and card renders visibly (not collapsed)

### CODE REVIEW

File: autogpt_platform/backend/backend/copilot/tools/ask_question.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        clarifying_question = ClarifyingQuestion(
+            question=question,
+            keyword=keyword,
+            example=example,
+        )
+        return ClarificationNeededResponse(
+            message=question,
+            session_id=session_id,
+            questions=[clarifying_question],
+        )
```
Comment (majdyz) [inline autogpt_platform/backend/backend/copilot/tools/ask_question.py:93]: 🤖 🟠 **Should Fix**: Missing colocated test file `ask_question_test.py`. The backend convention is to colocate `*_test.py` with source files. At minimum, test that `_execute` returns a `ClarificationNeededResponse` with the correct question/options/keyword mapping, and test the edge case where `options` is empty (example should be `None`).

File: autogpt_platform/backend/backend/copilot/tools/ask_question.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+import logging
+from typing import Any
+
+from backend.copilot.model import ChatSession
+
+from .base import BaseTool
+from .models import ClarificationNeededResponse, ClarifyingQuestion, ToolResponseBase
+
+logger = logging.getLogger(__name__)
```
Comment (majdyz) [inline autogpt_platform/backend/backend/copilot/tools/ask_question.py]: 🤖 🔵 **Nit**: `logger` is imported but never used — remove it.

File: autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/AskQuestion.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
+"use client";
+
+import { ChatTeardropDotsIcon, WarningCircleIcon } from "@phosphor-icons/react";
+import type { ToolUIPart } from "ai";
+import { ClarificationQuestionsCard } from "../CreateAgent/components/ClarificationQuestionsCard";
```
Comment (majdyz) [inline autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/AskQuestion.tsx]: 🤖 🟡 **Nice to Have**: `ClarificationQuestionsCard` is imported from `CreateAgent/components/` — this creates a coupling from a generic tool to an agent-specific tool's internal component. Consider lifting `ClarificationQuestionsCard` to a shared location (e.g. `copilot/components/`) so both `CreateAgent`, `EditAgent`, and `AskQuestion` import from the same shared component.

File: autogpt_platform/backend/backend/copilot/tools/ask_question.py (hunk 2 above)
Comment (majdyz) [reply autogpt_platform/backend/backend/copilot/tools/ask_question.py]: 🤖 Fixed in 48b849934f: Removed unused `logger` import.

File: autogpt_platform/backend/backend/copilot/tools/ask_question.py (hunk 1 above)
Comment (majdyz) [reply autogpt_platform/backend/backend/copilot/tools/ask_question.py:93]: 🤖 Fixed in 48b849934f: Added `ask_question_test.py` with 3 tests covering options, no-options, and keyword-only cases. All pass.

File: autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/AskQuestion.tsx (hunk 3 above)
Comment (majdyz) [reply autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/AskQuestion.tsx]: 🤖 Fixed in 48b849934f: Moved `ClarificationQuestionsCard` to shared `copilot/components/ClarificationQuestionsCard/`. AskQuestion now imports from there. CreateAgent re-exports from the shared location for backward compat.

File: autogpt_platform/frontend/src/app/(platform)/copilot/tools/CreateAgent/components/ClarificationQuestionsCard.tsx [hunk 4]
```diff
@@ -1,235 +1 @@
...
-      </div>
-    </div>
-  );
-}
-
-function getStorageKey(sessionId?: string): string | null {
-  if (!sessionId) return null;
-  return `clarification_answers_${sessionId}`;
-}
+export { ClarificationQuestionsCard } from "../../../components/ClarificationQuestionsCard/ClarificationQuestionsCard";
```
Comment (majdyz) [inline autogpt_platform/frontend/src/app/(platform)/copilot/tools/CreateAgent/components/ClarificationQuestionsCard.tsx]: 🤖 🔵 **Nit**: The re-export shim here works but the frontend guidelines say "No barrel files or `index.ts` re-exports". Since both `CreateAgent.tsx` and `EditAgent.tsx` import from this path, update them both to import directly from the shared location (`../../components/ClarificationQuestionsCard/ClarificationQuestionsCard`) and delete this re-export file entirely.

Comment (majdyz) [reply autogpt_platform/frontend/src/app/(platform)/copilot/tools/CreateAgent/components/ClarificationQuestionsCard.tsx]: 🤖 Fixed in 2f50facfa9: Updated both `CreateAgent.tsx` and `EditAgent.tsx` to import directly from the shared location. Deleted the re-export shim file.

File: autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/helpers.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    const obj = output as Record<string, unknown>;
+    if (
+      obj.type === ResponseType.agent_builder_clarification_needed ||
+      "questions" in obj
+    ) {
+      return obj as unknown as AskQuestionOutput;
+    }
+    if (obj.type === "error" || "error" in obj) {
+      return obj as unknown as ErrorOutput;
+    }
```
[the remark this answers is not in the document]
Comment (majdyz) [reply autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/helpers.ts:44]: 🤖 Fixed in ffa5a5b0a7: Aligned `isErrorOutput` to also check `"error" in output`, matching `parseOutput` logic.

File: autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/AskQuestion.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+  function handleAnswers(answers: Record<string, string>) {
+    if (!output || !isClarificationOutput(output)) return;
+    const questions = output.questions ?? [];
+    const message = questions
+      .map((q) => {
+        const answer = answers[q.keyword] || "";
+        return `> ${q.question}\n\n${answer}`;
+      })
+      .join("\n\n");
```
[the remark this answers is not in the document]
Comment (majdyz) [reply autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/AskQuestion.tsx:38]: 🤖 Fixed in 5706c78341: `handleAnswers` now uses `normalizeClarifyingQuestions()` so keyword lookups match the normalized keys from the card.

File: autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/AskQuestion.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    onSend(`**Here are my answers:**\n\n${message}\n\nPlease proceed.`);
```
[the remark this answers is not in the document]
Comment (majdyz) [reply autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/AskQuestion.tsx:39]: 🤖 Fixed in 5706c78341: `handleAnswers` now iterates `normalizeClarifyingQuestions(output.questions)` instead of raw `output.questions`, so normalized keywords align with the answer keys.

File: autogpt_platform/backend/backend/copilot/tools/ask_question.py [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    async def _execute(
+        self,
+        user_id: str | None,
+        session: ChatSession,
+        **kwargs: Any,
+    ) -> ToolResponseBase:
+        del user_id  # unused; required by BaseTool contract
+        question: str = kwargs.get("question", "")
+        options: list[str] = kwargs.get("options", [])
+        keyword: str = kwargs.get("keyword", "")
```
[the remark this answers is not in the document]
Comment (majdyz) [reply autogpt_platform/backend/backend/copilot/tools/ask_question.py]: 🤖 Fixed in 078e89f8a6: Added validation — empty/whitespace-only `question` now raises `ValueError`. Added test covering both cases.

File: autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/AskQuestion.tsx [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  if (output && isClarificationOutput(output)) {
+    return (
+      <ClarificationQuestionsCard
+        questions={normalizeClarifyingQuestions(output.questions ?? [])}
+        message={output.message}
```
[the remark this answers is not in the document]
Comment (majdyz) [reply autogpt_platform/frontend/src/app/(platform)/copilot/tools/AskQuestion/AskQuestion.tsx:46]: Fixed in 988edd6: Added the missing `sessionId={output.session_id}` prop to `ClarificationQuestionsCard` in `AskQuestion.tsx`. The previous commit (9990e9e) was about prompting_test, not this.

**Review by ntindle — APPROVED** [review]
